### ISSUE TITLE
508-defect-2 [FOCUS MANAGEMENT]: Back to Top Component - focus must be set on the body on click, and removed on blur

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#18426]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a keyboard user, I want focus to be returned to the top of the page when I click or press the Back to top button. It could be confusing and disorienting if I press the `TAB` key and focus jumps back to the bottom of the page.

## Details

The [Back to top JS file](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/btt-trevor/packages/formation/js/back-to-top.js) that controls scroll behavior needs to set a `tabindex="-1"` on the body tag on line 9 before calling `el.focus()`. Otherwise the back to top button will not release focus to the body tag. We should also set an event listener to remove the `tabindex` when the body tag loses focus.

I wrote a quick Codepen to illustrate the behavior and so we could easily test with multiple browsers: https://codepen.io/tpierce_402/pen/eYdPEjE

## Acceptance Criteria

- [ ] Focus is properly set on the `<body>` tag when users click or press Back to top
- [ ] The body tag should not have any visible focus halo
- [ ] Users should be able to set focus on the first link or button from the top of the page after pressing `TAB`. The page should not jump back to the Back to top button or scroll down.

## WCAG or Vendor Guidance (optional)

* [Meaningful Sequence: Understanding SC 1.3.2](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-sequence.html)

**Comment by bkjohnson** [comment]
Whether or not the halo is applied to the body is not the concern of the Back to Top component, in my opinion. That's up to the other styling on the page, and right now we know that that halo gets applied by default to everything that's focused and has a `tabindex`. However, I did check what this would look like on va.gov, and the halo is present but not visible. In order to get this screenshot I gave the `<body>` some margin so that the halo had the room to be visible:

![image](https://user-images.githubusercontent.com/2008881/104666898-cfb22900-5689-11eb-811d-d73726f0dc6a.png)

---

Here is what the update looks like:

When I click "Back to top":

1. We scroll to the top
1. The `<body>` gets a `tabindex` of -1

When I hit `Tab`:

1. The `tabindex` is removed from the body
1. The link containing "the introduction" gets focused as you can see from the yellow halo


![Peek 2021-01-14 16-51](https://user-images.githubusercontent.com/2008881/104666509-ea37d280-5688-11eb-9886-a44d4eec0e70.gif)

**Comment by bkjohnson** [comment]
Fixed in https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/562 according to a [Slack thread](https://dsva.slack.com/archives/C01DBGX4P45/p1611008184002800?thread_ts=1610125544.020700&cid=C01DBGX4P45)
